core/rend/gui.cpp

```diff
@@ -1933,6 +1931,45 @@ static void gui_display_loadscreen()
     ImGui::End();
 }
 
+void gui_select_save_states(bool saving) {
+    centerNextWindow();
+    ImGui::SetNextWindowSize(ImVec2(330 * scaling, 0));
+
+    ImGui::Begin("##select states", NULL, ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoMove | ImGuiWindowFlags_AlwaysAutoResize);
+
+	ImGui::Columns(2, "buttons", false);
+
+	for (int i = 0; i < SAVE_STATES_CAPACITY; i++) {
+		if (i != 0)
+			ImGui::NextColumn();
+
+		char state_button_name[32];
+		sprintf(state_button_name, "State %d", i);
+		if (ImGui::Button(state_button_name, ImVec2(150 * scaling, 50 * scaling)))
+		{
+			gui_state = GuiState::Closed;
+			if (saving)
+			{
+				gui_state = GuiState::Closed;
+				dc_savestate(i);
+			}
+			else
+			{
+				gui_state = GuiState::Closed;
+				dc_loadstate(i);
+			}
+		}
+	}
+
+	ImGui::NextColumn();
+	if (ImGui::Button(saving ? "Stop Saving" : "Stop Loading", ImVec2(150 * scaling, 50 * scaling)))
```

Comment:
I would simply name this button "Cancel" instead of Stop Saving/Loading.
And centering the button horizontally would be nice too.